File: WebShop_FSharp/Middleware/CitiesMiddleware.fs
namespace WebShop_FSharp.Middleware

open System.Collections.Generic
open DomainModels
open Microsoft.AspNetCore.Http
open System.Linq

type CitiesMiddleware(next: RequestDelegate) =
    let mutable cities: List<string> = null
    member this.Next = next
    
    member this.InvokeAsync(context: HttpContext, dbContext: ApplicationContext) =
        cities ??= dbContext.Cities.Select(x => x.Name).ToList();
		context.Items["cities"] = cities;
		next.Invoke(context);

File: WebShop_NULL/Middleware/CitiesMiddleware.cs
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using DomainModels;
using Microsoft.AspNetCore.Http;

namespace WebShop_NULL.Middleware
{
	public class CitiesMiddleware
	{
		private readonly RequestDelegate next;
		private List<string> cities;
		
		public CitiesMiddleware(RequestDelegate next)
		{
			this.next = next;
		}
		
		public async Task InvokeAsync(HttpContext context, ApplicationContext dbContext)
		{
			cities ??= dbContext.Cities.Select(x => x.Name).ToList();
			context.Items["cities"] = cities;
			await next.Invoke(context);
		}
	}
}